client/src/api/categoriesApi.js

```diff
@@ -0,0 +1,19 @@
+import { extractData } from './helpers';
+import request from './request';
+
+const urls = {
+    root: '/categories',
+    get categories() {
+      return this.root + '/list';
+    },
+  };
+  
+  
+  const categories = () => {
```

Comment:
function name thing again. The point of this is that you import it somewhere and it looks like this
```js
categoriesApi.categories // the categories thing here is redundant
```
instead, it would look like this
```js
categoriesApi.list // makes more sense
```
this would be even better
```js
categoriesApi.fetchAll // because now we understand what it does entirely from any context - we are calling categories api and fetching all our categories
```

Reply:
same naming scheme goes for all other apis you created in this PR